src/main/scala/census/db/creator/service/actors/DownloadActor.scala

```diff
@@ -0,0 +1,67 @@
+package census.db.creator.service.actors
+import java.io.File
+import java.nio.file.Paths
+import java.util.concurrent.atomic.AtomicInteger
+
+import akka.actor.{Actor, ActorRef, ActorSystem}
+import akka.http.scaladsl.Http
+import akka.http.scaladsl.model.HttpRequest
+import akka.stream.Materializer
+import akka.stream.scaladsl.FileIO
+import census.db.creator.config.Config
+
+import scala.collection.mutable
+import scala.concurrent.ExecutionContext
+import scala.util.Failure
+
+case class DownloadMessage(url: String)
+
+private case object CheckQueueMessage
+
+class DownloadActor(config: Config, unzipActor: ActorRef)(
+  private implicit val executionContext: ExecutionContext,
+  private implicit val materializer: Materializer,
+) extends Actor {
+  private implicit val actorSystem: ActorSystem = context.system
+
+  import scala.concurrent.duration._
+  actorSystem.scheduler.schedule(Duration.Zero, 1.second, self, CheckQueueMessage)
+
+  private val filesToDownload = mutable.Queue[String]()
+  private val downloadingsInProgress = new AtomicInteger(0)
+
+  override def receive: Receive = {
+    case DownloadMessage(url) =>
+      filesToDownload += url
+    case CheckQueueMessage =>
+      while (downloadingsInProgress.get() < 20 && filesToDownload.nonEmpty) {
```

Comment:
This is simple enough that it is probably ok here. For the future maybe think of spinning up 20 actors for fanning the download. Then you would't have to worry about the AtomicInteger. Anytime you start thinking about concurrency when using actors then you are likely in need of another hierarchy in your actor system. If this actor weren't closed by System.exit in its usage here then this scheduler would be unnecessary when it empties and should really shut itself  down. Again, with an actor approach you would only need a scheduler to check for dead downloads. But even that could probably be handled otherwise. 

TLDR - it's fine, and here are some future tips :)